packages/block-library/src/gallery/index.php

```diff
@@ -81,9 +81,23 @@ function block_core_gallery_render( $attributes, $content ) {
 	}
 
 	// Set the CSS variable to the column value, and the `gap` property to the combined gap value.
-	$style = '.wp-block-gallery.' . $class . '{ --wp--style--unstable-gallery-gap: ' . $gap_column . '; gap: ' . $gap_value . '}';
+	$gallery_styles   = array();
+	$gallery_styles[] = array(
+		'selector'     => ".wp-block-gallery.{$class}",
+		'declarations' => array(
+			'--wp--style--unstable-gallery-gap' => $gap_column,
+			'gap'                               => $gap_value,
+		),
+	);
+
+	gutenberg_style_engine_get_stylesheet_from_css_rules(
```

Comment:
I think we might have to come up with a way to ensure these styles are printed after the layout styles

- https://github.com/WordPress/gutenberg/pull/41423
- 
cc @glendaviesnz

Reply:
Following the testing in #41423, it looks like the styling rules aren't being output on a classic theme for me (TwentyTwentyOne), so if I add the following to my theme's css, it doesn't affect the gallery gap:

```
html {
	--wp--style--gallery-gap-default: 37px;
}
```

From a bit of introspection, it appears that the fallback rules in `$fallback_gap` are being stripped in WP 6.0, as it's a pretty complex value, and the `gutenberg_safecss_filter_attr_allow_css_6_1` function [here](https://github.com/WordPress/gutenberg/blob/7d99c49fd267de4c1cccda0224f35a20c8e887f6/lib/compat/wordpress-6.1/blocks.php#L39) hasn't been updated to reflect the final version in 6.1 on these lines: https://github.com/WordPress/wordpress-develop/blob/2b4d385298504d412e9f0dea2e7019d53463c705/src/wp-includes/kses.php#L2508-L2512

If I copy over that `preg_replace` line to the Gutenberg function, it appears that the style output works correctly, so we might need to land another backport of a backport before we can go with this PR?

Reply:
I'll put up a PR to backport that line.